app/services/hashtag_autocomplete_service.rb

```diff
@@ -279,6 +281,23 @@ def lookup_old(slugs)
 
   private
 
+  def search_without_term(types_in_priority_order, limit)
+    split_limit = (limit.to_f / types_in_priority_order.length.to_f).ceil
+    limited_results = []
+
+    types_in_priority_order.each do |type|
+      search_results = @@data_sources[type].search_without_term(guardian, split_limit)
+      next if search_results.empty?
+      next if !all_data_items_valid?(search_results)
+
+      # This is purposefully unsorted as search_without_term should sort
+      # in its own way.
+      limited_results.concat(set_types(set_refs(search_results), type))
+    end
+
+    limited_results.take(limit)
```

Comment:
I just noticed we only apply the limit here instead of within the loop. I wonder if it makes sense to terminate the loop early if we already have sufficient results. Otherwise, I think we will always be executing multiple queries each time?

Reply:
I only do this here because `split_limit` may end up with an uneven number that is over the acutal limit (e.g. 21 instead of 20) so this just makes sure it meets the cutoff -- individual queries will not return > than the split limits provided (which in the case of `20` is `7` each).